### PR TITLE
Component/tertiary button

### LINKED ISSUE #8: TertiaryButton
Build the TertiaryButton component.

Screenshot 2020-12-13 at 21 15 46

View the component in [Figma](https://www.figma.com/file/gGwSe7ilZCUmAIaJyUc4iY/Bristol-Tech-Fair-Design-System?node-id=489%3A43).

### PR DESCRIPTION
**Pull request by merryface**
#8 
1. Created button with correct styling

2. copied example component for props:
-   type = submit/button (defaulted to button)
-   onClick
-   content

3. added prop of "colour". defaulted to 'blue.' any other value will turn it to the alternate colour

### CODE REVIEW

File: src/components/TertiaryButton/TertiaryButton.jsx
Comment (RobTF9) [inline src/components/TertiaryButton/TertiaryButton.jsx:13]: Change wrapper to button component.

File: src/components/TertiaryButton/TertiaryButton.styles.js
Comment (RobTF9) [inline src/components/TertiaryButton/TertiaryButton.styles.js:7]: 1px > 0.1rem

File: src/components/TertiaryButton/TertiaryButton.styles.js
Comment (RobTF9) [inline src/components/TertiaryButton/TertiaryButton.styles.js:4]: Double check paddding matches designs.

File: src/components/TertiaryButton/TertiaryButton.styles.js
Comment (RobTF9) [inline src/components/TertiaryButton/TertiaryButton.styles.js:11]: 100px > 10rem

File: src/components/TertiaryButton/TertiaryButton.styles.js
Comment (RobTF9) [inline src/components/TertiaryButton/TertiaryButton.styles.js:10]: Remove box-sizing, applied by default

File: src/components/TertiaryButton/TertiaryButton.test.jsx
Comment (RobTF9) [inline src/components/TertiaryButton/TertiaryButton.test.jsx:12]: Test click handler

**Review by RobTF9 — CHANGES_REQUESTED** [review]

**Comment by merryface** [comment]
fixed all issues as pointed out by first review. Tests were copied from Primary Button

1. Added additional styling to be more consistent with primary button while respecting the figma design
2. have no idea how to run tests (never used jest or anything like that before)

**Comment by shanesc** [comment]
> fixed all issues as pointed out by first review. Tests were copied from Primary Button
> 
> 1. Added additional styling to be more consistent with primary button while respecting the figma design
> 2. have no idea how to run tests (never used jest or anything like that before)
> 
> was unsure if I should use medium or large lateral padding. Picked the closest looking one (medium)

Padding looks good.
FYI if you hover of a component in Figma you can see distances from other elements:
image

File: src/components/TertiaryButton/TertiaryButton.stories.jsx
Comment (shanesc) [inline src/components/TertiaryButton/TertiaryButton.stories.jsx:10]: Remove `colour="not blue"`, not needed

File: src/components/TertiaryButton/TertiaryButton.styles.js
Comment (shanesc) [inline src/components/TertiaryButton/TertiaryButton.styles.js:7]: Delete the empty line break whitespace to be consistent with other CSS

File: src/components/TertiaryButton/TertiaryButton.styles.js
Comment (shanesc) [inline src/components/TertiaryButton/TertiaryButton.styles.js:9]: remove the `border-color` property and move `var(--light-blue)` up to the `border` property
i.e.
`border: 0.1rem solid var(--lightblue);`

File: src/components/TertiaryButton/TertiaryButton.jsx
Comment (shanesc) [inline src/components/TertiaryButton/TertiaryButton.jsx:16]: I asked @ammersive about this as well, what do you both think about having the `onClick` handler prop being required on buttons? That way there's less likelihood of a bug with a button not performing an action.

Happy with it either way we decide, just a thought.

File: src/components/TertiaryButton/TertiaryButton.test.jsx
Comment (shanesc) [inline src/components/TertiaryButton/TertiaryButton.test.jsx:29]: Tests look good!

**Review by shanesc — CHANGES_REQUESTED** [review]
Overall looks really good, just a couple of style changes to be made.

File: src/components/TertiaryButton/TertiaryButton.jsx
Comment (shanesc) [reply src/components/TertiaryButton/TertiaryButton.jsx:16]: Scratch that, not needed. Here was Rob's response:

> No, you don't necessarily need one. In forms for example, the event is handled my the submission of the form as opposed to the button so a onClick is not always required.

Comment (merryface) [reply src/components/TertiaryButton/TertiaryButton.jsx:16]: frankly i literally copied your tests for the primary button, I'm still studying them in my own time and not entirely comfortable as to how and when to do them, So I'm glad Rob said no need :P

**Review by shanesc — APPROVED** [review]
Looks good, co ahead and click "Squash and Merge" 👍 

*edit: just saw that you can't merge until @RobTF9 and I both approve the reviews*

**Review by RobTF9 — COMMENTED** [review]
Looks good to me 👍

**Review by RobTF9 — APPROVED** [review]

**Comment by merryface** [comment]
Rob is the workflow to delete the branch afterwards?

**Comment by RobTF9** [comment]
Yep

**Pull request merged**
